File: OpenMediationDemo/InterstitialViewController.m
//
// Copyright 2020 ADTIMING TECHNOLOGY COMPANY LIMITED
// Licensed under the GNU Lesser General Public License Version 3

#import "InterstitialViewController.h"

@implementation InterstitialViewController

- (void)viewDidLoad {
    self.title = @"Interstitial";
    self.adFormat = OpenMediationAdFormatInterstitial;
    [super viewDidLoad];
}

- (void)loadAd {
    [[OMInterstitial sharedInstance] addDelegate:self];
    if ([[OMInterstitial sharedInstance]isReady]) {
        self.showItem.enabled = YES;
    }
}


-(void)showItemAction {
    [self.view endEditing:YES];
    NSLog(@"isCappedForScene:%@,%d",self.loadID,[[OMInterstitial sharedInstance]isCappedForScene:self.loadID]);
    [[OMInterstitial sharedInstance]showWithViewController:self scene:self.loadID];
}

#pragma mark -- InterstitialDelegate

/// Invoked when a interstitial video is available.
- (void)omInterstitialChangedAvailability:(BOOL)available {
    self.showItem.enabled = available;
    if (available) {
         [self showLog:@"Interstitial ad did load"];
    }
}

/// Sent immediately when a interstitial video is opened.
- (void)omInterstitialDidOpen:(OMScene*)scene {
        [self showLog:@"Interstitial ad open"];
}

/// Sent immediately when a interstitial video starts to play.
- (void)omInterstitialDidShow:(OMScene*)scene {
        [self showLog:@"Interstitial ad show"];
}

/// Sent after a interstitial video has been clicked.
- (void)omInterstitialDidClick:(OMScene*)scene {
        [self showLog:@"Interstitial ad click"];
}

/// Sent after a interstitial video has been closed.
- (void)omInterstitialDidClose:(OMScene*)scene {
        [self showLog:@"Interstitial ad close"];
}

/// Sent after a interstitial video has failed to play.
- (void)omInterstitialDidFailToShow:(OMScene*)scene withError:(NSError *)error {
        [self showLog:@"Interstitial fail to show"];
}

@end

File: OpenMediationSwiftDemo/InterstitialViewController.swift
// Copyright 2020 ADTIMING TECHNOLOGY COMPANY LIMITED
// Licensed under the GNU Lesser General Public License Version 3

import UIKit
import OpenMediation
class InterstitialViewController: UIViewController, OMInterstitialDelegate {
    
    var logLabel = UILabel()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "Interstitial"
        self.view.backgroundColor = UIColor.white
        
        let showBtn:UIButton = UIButton(frame: CGRect(x: 20, y: 120, width: 120, height: 30))
        showBtn.contentHorizontalAlignment = UIControl.ContentHorizontalAlignment.left
        showBtn.backgroundColor = UIColor.white
        showBtn.setTitleColor(UIColor.blue, for: .normal)
        showBtn.setTitleColor(UIColor.gray, for: .highlighted)
        showBtn.setTitleColor(UIColor.gray, for: .selected)
        showBtn.titleLabel?.font = UIFont.systemFont(ofSize: 17)
        showBtn.setTitle("Show", for: .normal)
        showBtn.addTarget(self, action: #selector(showBtnDidClicked), for: .touchUpInside)
        self.view.addSubview(showBtn)
        
        logLabel = UILabel(frame: CGRect(x: 20, y: showBtn.frame.maxY+20, width: 200, height: 30))
        logLabel.text = "Log"
        logLabel.font = UIFont.systemFont(ofSize: 15)
        self.view.addSubview(logLabel)
        
        OMInterstitial.sharedInstance().add(self);
    }
    
    // Scene is optional，if you don't want to use it just ignore the sceneName parameter or use value "". 
    @objc func showBtnDidClicked(){
        if OMInterstitial.sharedInstance().isReady() {
            OMInterstitial.sharedInstance().show(with: self, scene: "YOUR_SCENE_NAME")
        }
    }
    
    /// Invoked when interstitial video is available.
    func omInterstitialChangedAvailability(_ available: Bool) {
        if available {
            print("InterstitialAd is Available")
            logLabel.text = "ad available"
        }else{
            logLabel.text = "ad invailable"
        }
    }
    
    /// Sent immediately when a interstitial video is opened.
    func omInterstitialDidOpen(_ scene: OMScene) {
        logLabel.text = "InterstitialAd is Open"
        print("InterstitialAd is Open")
    }
    
    /// Sent immediately when a interstitial video starts to play.
    func omInterstitialDidShow(_ scene: OMScene) {
        logLabel.text = "InterstitialAd Start Play"
        print("InterstitialAd Start Play")
    }
    
    /// Sent after a interstitial video has been clicked.
    func omInterstitialDidClick(_ scene: OMScene) {
        logLabel.text = "InterstitialAd Did Click"
        print("InterstitialAd Did Click")
    }
    
    /// Sent after a interstitial video has been closed.
    func omInterstitialDidClose(_ scene: OMScene) {
        logLabel.text = "InterstitialAd Did Close"
        print("InterstitialAd Did Close")
    }
    
    /// Sent after a interstitial video has failed to play.
    func omInterstitialDidFail(toShow scene: OMScene, withError error: Error) {
        logLabel.text = "InterstitialAd failed to play"
        print("InterstitialAd failed to play")
    }
}
